backend/internal/interface/api/handler/hello_world_handler_test.go

```diff
@@ -0,0 +1,131 @@
+package handler
+
+import (
+	"myapp/internal/application/usecase"
+	"myapp/internal/domain/model"
+	"myapp/internal/domain/repository/repository_mock"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+)
+
+func TestNewHelloWorldHandler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockRepo := repository_mock.NewMockHelloWorldRepository(ctrl)
+	mockUsecase := usecase.NewHelloWorldUsecase(mockRepo)
+
+	type args struct {
+		u usecase.HelloWorldUsecase
+	}
+	tests := []struct {
+		name string
+		args args
+		want HelloWorldHandler
+	}{
+		{
+			name: "success",
+			args: args{
+				u: mockUsecase,
+			},
+			want: HelloWorldHandler{
+				u: mockUsecase,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewHelloWorldHandler(tt.args.u); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewHelloWorldHandler() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestHelloWorldHandler_HelloWorld(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	mockRepo := repository_mock.NewMockHelloWorldRepository(ctrl)
+	mockUsecase := usecase.NewHelloWorldUsecase(mockRepo)
+	handler := NewHelloWorldHandler(mockUsecase)
+
+	tests := []struct {
+		name           string
+		lang           string
+		mockReturn     *model.HelloWorld
+		mockError      error
+		expectedStatus int
+		expectedBody   string
+	}{
+		{
+			name:           "success",
+			lang:           "en",
+			mockReturn:     &model.HelloWorld{Message: "Hello, World!", Lang: "en"},
+			mockError:      nil,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"message":"Hello, World!","lang":"en"}`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.mockError == nil {
+				mockRepo.EXPECT().Get(gomock.Any(), tt.lang).Return(tt.mockReturn, nil).AnyTimes()
+			} else {
+				mockRepo.EXPECT().Get(gomock.Any(), tt.lang).Return(nil, tt.mockError).AnyTimes()
+			}
```

Comment:
分岐必要なかったりする？

あと、一回だけ呼ばれればいいからAnyTimesはいらないかも！

```suggestion
			mockRepo.EXPECT().Get(gomock.Any(), tt.lang).Return(tt.mockReturn, tt.mockError)
```

Reply:
確かにここは条件分岐不要だったので[1680a90](https://github.com/givery-bootcamp/dena-2024-team3/pull/96/commits/1680a9023e48478b70f738623b595050e80dd063)で削除した！